lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -0,0 +1,121 @@
+import path from "node:path";
+import os from "node:os";
+import AbstractResolver from "./AbstractResolver.js";
+import Installer from "./maven/Installer.js";
+import {getLogger} from "@ui5/logger";
+const log = getLogger("ui5Framework:Sapui5MavenSnapshotResolver");
+
+const DIST_PKG_NAME = "@sapui5/distribution-metadata";
+const DIST_GROUP_ID = "com.sap.ui5.dist";
+const DIST_ARTIFACT_ID = "sapui5-sdk-dist";
+
+/**
+ * Resolver for the SAPUI5 framework
```

Comment:
does this resolver has to be public? Its a completly sap internally used feature, isn't?

Reply:
Not quite sure. The argument here is that the rest of the resolvers are also public.

Reply:
Removed the jsdoc anotation. Would that be sufficient?

Reply:
Maybe I should have phrased my comment differently: I'm okay with having it public API. It could be used internally by other teams.